nifi-nar-bundles/nifi-mongodb-bundle/nifi-mongodb-processors/src/main/java/org/apache/nifi/processors/mongodb/PutMongo.java

```diff
@@ -86,19 +91,27 @@
         .name("Update Query Key")
         .description("Key name used to build the update query criteria; this property is valid only when using update mode, "
                 + "otherwise it is ignored")
-        .required(true)
-        .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+        .required(false)
+        .addValidator(Validator.VALID)
```

Comment:
Since required is set to false, you can still use the original validor: it'll be used only if a value is set. No? Unless you really want someone to be able to set an empty string as a possible value.

Reply:
Wasn't sure if that's the case.